src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ProcessTests.cs

```diff
@@ -145,7 +145,137 @@ public async Task MultiProcessIdentificationTest(DiagnosticPortConnectionMode mo
+                runner.DiagnosticPortPath = diagnosticPortPath;
+                runner.ScenarioName = TestAppScenarios.AsyncWait.Name;
+                runner.Environment[ExpectedEnvVarName] = Guid.NewGuid().ToString("D");
+
+                appRunners[i] = runner;
+            }
+
+            IEnumerable<ProcessIdentifier> identifiers;
+            await appRunners.ExecuteAsync(async () =>
+            {
+                // Query for process identifiers
+                identifiers = await apiClient.GetProcessesAsync();
+
+                Assert.NotNull(identifiers);
+
+                List<int?> pids = new List<int?>();
+                List<Guid?> uids = new List<Guid?>();
+                List<string> names = new List<string>();
+
+                foreach (ProcessIdentifier processIdentifier in identifiers)
+                {
+                    pids.Add(processIdentifier.Pid);
+                    uids.Add(processIdentifier.Uid);
+                    names.Add(processIdentifier.Name);
+                }
+
+#if NET5_0_OR_GREATER
+                // CHECK 1: Get response for processes using PID, UID, and Name and check for consistency
+
+                List<ProcessInfo> processInfoQueriesCheck1 = new List<ProcessInfo>();
+
+                processInfoQueriesCheck1.Add(await apiClient.GetProcessAsync(pid: pids[0], uid: null, name: null, System.Threading.CancellationToken.None));
+                processInfoQueriesCheck1.Add(await apiClient.GetProcessAsync(pid: null, uid: uids[0], name: null, System.Threading.CancellationToken.None));
+                //processInfoQueriesCheck1.Add(await apiClient.GetProcessAsync(pid: null, uid: null, name: names[0], System.Threading.CancellationToken.None));
+
+                VerifyProcessInfoEquality(processInfoQueriesCheck1);
+#endif
+                // CHECK 2: Get response for requests using PID | PID and UID | PID, UID, and Name and check for consistency
+
+                List<ProcessInfo> processInfoQueriesCheck2 = new List<ProcessInfo>();
+
+                processInfoQueriesCheck2.Add(await apiClient.GetProcessAsync(pid: pids[0], uid: null, name: null, System.Threading.CancellationToken.None));
+                processInfoQueriesCheck2.Add(await apiClient.GetProcessAsync(pid: pids[0], uid: uids[0], name: null, System.Threading.CancellationToken.None));
+                processInfoQueriesCheck2.Add(await apiClient.GetProcessAsync(pid: pids[0], uid: uids[0], name: names[0], System.Threading.CancellationToken.None));
+
+                VerifyProcessInfoEquality(processInfoQueriesCheck2);
+
+                // CHECK 3: Get response for processes using PID and an unassociated (randomly generated) UID and ensure the proper exception is thrown
+
+                await Assert.ThrowsAsync<ValidationProblemDetailsException>(async () => await apiClient.GetProcessAsync(pid: pids[0], uid: Guid.NewGuid(), name: null, System.Threading.CancellationToken.None));
```

Comment:
For each of these that we expect to throw ValidationProblemDetailsException, please assert the status code (I believe it is 400). You'll find examples on the other tests.